src/api.jl

```diff
@@ -0,0 +1,24 @@
+module api
+
+paths = ByteString[]
+const libvulkan = Libdl.find_library(["libvulkan", "vulkan", "libvulkan.so.1"], paths)
+@assert libvulkan != ""
+
+function _vk_func(func, ret_type, arg_types)
+  local args_in = Symbol[symbol("arg$i::$T")
+  for (i, T) in enumerate(arg_types.args)]
+  esc(quote
+    function $func($(args_in...))
+      ccall(($(string(func)), libvulkan),
+      $ret_type,
+      $arg_types,
+      $(args_in...))
+    end
+  end)
+end
```

Comment:
`Ptr` vs `Ref` is not the issue, just remove the constraint on the function arguments should be good enough. (you'll need to define the correct cconvert and unsafe_convert later to actually make use it)